sdk/eventhub/Azure.Messaging.EventHubs/tests/EventHubProducer/EventHubProducerLiveTests.cs

```diff
@@ -181,5 +185,308 @@ await using (var producer = client.CreateProducer())
+        {
+            await using (var scope = await EventHubScope.CreateAsync(1))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                await using (var client = new EventHubClient(connectionString))
+                await using (var producer = client.CreateProducer())
+                {
+                    var singleEvent = new EventData(new byte[0]);
+                    var eventBatch = new[] { new EventData(new byte[0]) };
+
+                    Assert.That(async () => await producer.SendAsync(singleEvent), Throws.Nothing);
+                    Assert.That(async () => await producer.SendAsync(eventBatch), Throws.Nothing);
+                }
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="EventHubProducer" /> is able to
+        ///   connect to the Event Hubs service and perform operations.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ProducerCanSendSingleLargeEvent()
+        {
+            await using (var scope = await EventHubScope.CreateAsync(1))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                await using (var client = new EventHubClient(connectionString))
+                await using (var producer = client.CreateProducer())
+                {
+                    // Actual limit is 1046520 for a single event
+                    var singleEvent = new EventData(new byte[1000000]);
+                    var eventBatch = new[] { new EventData(new byte[1000000]) };
+
+                    Assert.That(async () => await producer.SendAsync(singleEvent), Throws.Nothing);
+                    Assert.That(async () => await producer.SendAsync(eventBatch), Throws.Nothing);
+                }
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="EventHubProducer" /> is able to
+        ///   connect to the Event Hubs service and perform operations.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ProducerCannotSendSingleEventLargerThanMaximumSize()
+        {
+            await using (var scope = await EventHubScope.CreateAsync(1))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                await using (var client = new EventHubClient(connectionString))
+                await using (var producer = client.CreateProducer())
+                {
+                    // Actual limit is 1046520 for a single event
+                    var singleEvent = new EventData(new byte[1500000]);
+                    var eventBatch = new[] { new EventData(new byte[1500000]) };
+
+                    Assert.That(async () => await producer.SendAsync(singleEvent), Throws.TypeOf<TrackOne.MessageSizeExceededException>());
```

Comment:
I need to capture and re-wrap exceptions to avoid exposing the TrackOne types.   I'll see if I have time to address `MessageSizeExceededException` a for the first preview.